pallets/inflation-manager/src/lib.rs

```diff
@@ -180,28 +171,34 @@ pub mod pallet {
 			let inflation_config = InflationConfiguration::<T>::get();
 			let mut inflation_parameters = InflationParameters::<T>::get();
 
-			// if we're at the end of a year
+			// if we're at the end of a year or initializing inflation
 			if now >= target_block {
 				// update current year
 				CurrentYear::<T>::put(new_year);
 
-				// check if we need to recalculate inflation parameters for a new year
-				// update inflation parameters if we havent reached the stagnation year
+				// if we're at DoInitializeAt, then we need to adjust total issuance for delayed TGE
+				if now == T::DoInitializeAt::get() {
+					Self::fund_difference_balances();
+				}
+
+				// if we're not at the stagnation year, update inflation parameters
 				if new_year < inflation_config.inflation_stagnation_year {
 					// update inflation parameters
 					inflation_parameters = Self::update_inflation_parameters(&inflation_config);
 				}
 
-				// if, at end of year, we have reached the stagnation year, kill the recalculation
-				// flag and set inflation parameters to stagnation values
+				// if we're at the stagnation year, set inflation parameters to the stagnation rate
 				if new_year == inflation_config.inflation_stagnation_year {
 					inflation_parameters = InflationParametersT {
 						inflation_rate: inflation_config.inflation_stagnation_rate,
 						disinflation_rate: Perbill::one(),
 					};
 				}
 
-				InflationParameters::<T>::put(inflation_parameters.clone());
+				// no need to update InflationParameters in storage after inflation stagnation year
+				if current_year <= inflation_config.inflation_stagnation_year {
```

Comment:
After checking this logic, I felt maybe we can simply it by below one of three ways
1. Integrate this logic into the above condition
2. Even when `new_year ==  inflation_config.inflation_stagnation_year` or `new_year >=  inflation_config.inflation_stagnation_year`, we still write the configuration into the database
3. Or using new_year or current_year in these three conditions

Do you think it's more readable?

Reply:
Once we hit stagnation, InflationParameters will remain constant at 1% and Disinflation at 100%(meaning no disinflation), so once we reach stagnation year, we put these parameters into storage, and don't need to update them anymore, which is why this condition is here.

I think I understood your concern correctly, lemme know otherwise.

Reply:
ohhh i get it now, I think you located a bug, lemme test it out

Reply:
This is now optimized as a if..else if code block.